userland/userland_generic.ld

```diff
@@ -66,95 +66,79 @@ SECTIONS {
         KEEP (*(.syscalls))
         _etext = .;
         *(.ARM.extab*)
+        . = ALIGN(4); /* Make sure we're word-aligned here */
     } > FLASH =0xFF
 
 
-/* App state section. Used for persistent app data. */
+    /* App state section. Used for persistent app data. */
     .app_state :
     {
         KEEP (*(.app_state))
     } > FLASH =0xFF
 
-/* Beginning of SRAM */
-    _sram_start = .;
-
-/* Global Offset Table */
+    /* Global Offset Table */
     .got :
     {
         _got = .;
         *(.got*)
-        _egot = .;
-        _plt = .;
         *(.got.plt*)
-        _eplt = .;
     } > SRAM AT > FLASH
 
-/* Data section, static initialized variables
- *  Note: This is placed in Flash after the text section, but needs to be
- *  moved to SRAM at runtime
- */
+    /* Data section, static initialized variables
+     *  Note: This is placed in Flash after the text section, but needs to be
+     *  moved to SRAM at runtime
+     */
     .data :
     {
         _data = .;
         KEEP(*(.data*))
-        _edata = .;
     } > SRAM AT > FLASH
 
-/* BSS section, static uninitialized variables */
-    _sbss = .;
+    /* BSS section, static uninitialized variables */
     .bss :
     {
         _bss = .;
         KEEP(*(.bss*))
         *(COMMON)
-        _ebss = .;
     } > SRAM
 
-/*
- * __NOTE__: The following symbols are used only to pass information
- * through the elf -> tbf -> Tock kernel.
- *
- * The kernel will place the stack at the beginning of the SRAM section so
- * that stack overflows run off the end of the memory segment and trigger an
- * MPU violation instead of overwriting data/got/bss information. This means
- * the actual location of symbols in those sections in memory will be offset
- * by STACK_SIZE.
```

Comment:
This is still true (`s/kernel/crt0` perhaps), isn't it?

Reply:
It is, but is it really relevant? crt0 can actually still place the stack wherever it wants. This part of the linker script has no impact on it.

Reply:
I suppose it's less relevant once all the variables are deleted as well. The comment in part was to warn away folks from trying to use these symbols for anything meaningful at runtime -- deleting them completely probably achieves that better anyway :)